core/container/build_registry.go

```diff
@@ -35,6 +35,21 @@ func (br *BuildRegistry) BuildStatus(ccid string) (*BuildStatus, bool) {
 	return bs, ok
 }
 
+// ResetBuildStatus returns a new BuildStatus for the ccid, and false,
+// indicating that this build status is their responsibility. The caller
+// must use external locking to ensure the build status is not reset by
+// another install request and must call Notify with the error (or nil)
+// upon completion.
+func (br *BuildRegistry) ResetBuildStatus(ccid string) (*BuildStatus, bool) {
+	br.mutex.Lock()
+	defer br.mutex.Unlock()
+
+	bs := NewBuildStatus()
+	br.builds[ccid] = bs
+
+	return bs, false
```

Comment:
Why return a constant false here?

Reply:
To incorporate resetting the status into the existing flow. 

We get the build status, if ok (aka build status already exists), we then check for an error and reset the build status. We then continue to the existing logic of: if not ok, build the chaincode (which is also still applicable for the initial install/build of a chaincode).

Reply:
If this is not satisfying an interface, which I don't think it is, I can't think of a good reason to return a parameter which is a fixed constant, and has no obvious way to ever change.  Essentially, we are modifying the signature of a function in another package in a non-obvious way in order to make the code look simpler, which doesn't sit well with me.

If you really want to integrate it into the existing flow, you can always do something like:

```
buildStatus, ok = br.ResetBuildStatus(), false
```

Though I'd think it would be more readable over two lines.  [edit: or, looking at that logic, I think it's simpler to ignore the 'ok' variable, as its only possible value is false when it's next checked anyway]

Reply:
Fixed.